### ISSUE TITLE
Menu Item when using React Router Link: PropType error and aggressive styling 

### ISSUE THREAD

**Issue by cstrat** [segmentio/evergreen#728]
One issue and one annoyance.

1 - `Warning: Failed prop type: Invalid prop 'is' supplied to 'MenuItem'`
2 - Doing this with a Menu.Item renders the menu with an underline and coloured like an anchor. Really it should still be styled as a menu-item and not a link. I can override it manually but thats a pain.

Code:

```<Menu>
  <Menu.Item
    is={Link}
    to="/settings/"
    icon="settings"
  >
    Settings
  </Menu.Item>
</Menu>
```

Image:

![image](https://user-images.githubusercontent.com/4209076/73590889-1b80c680-453c-11ea-8477-5e1d4e8006c5.png)

**Comment by Chrischuck** [comment]
For your first issue, looks like the `Menu.Item` component is missing `PropTypes.elementType` in its propTypes validation. 

For the second issue, I would suggest to pass your own `className` to `Menu.Item`. I think it's best not to try an have the library override another library's behavior and leave that up to the developer to change. 

Thoughts @mshwery ?

**Comment by cstrat** [comment]
Yeah the second point I am not wasn't sure of however in other parts of Evergreen it is overridden. Check the docs under 'Button'.
For consistency I expected Menu.Item to behave the same.

![image](https://user-images.githubusercontent.com/4209076/73624761-2cdde600-4696-11ea-9655-371061cb5718.png)

**Comment by mshwery** [comment]
@cstrat @Chrischuck I think it's a reasonable expectation, given that menu is likely to include anchor tags. `Button` [has code that expects anchor tags may be used](https://github.com/segmentio/evergreen/blob/78cf402525b59fcac49e53236f382dba0c60256f/src/themer/src/createButtonAppearance.js#L10). 

Can you use `ui-box` css props to override the style: e.g. `textDecoration="none"` in the meantime?

**Comment by Chrischuck** [comment]
Nice, I can fix that tonight

**Comment by cstrat** [comment]
@mshwery @Chrischuck thanks guys, that is great.

I wanted to try look into how I could try doing a PR myself but I've never used typescript so I'd end up making a mess of it all 😃 

I am overriding the styles at the moment, cheers for your help 👏

**Comment by Chrischuck** [comment]
@cstrat You shouldn't have to mess with `ts` all the src code is in `js`! It's a little tricky to figure out how the theming/styling works under the hood though.

**Comment by cstrat** [comment]
@Chrischuck Oh I thought it was `ts` - I must be confused with something else. I'll take another look this weekend 😄

Thanks again for jumping on the issue so quickly.

**Comment by cstrat** [comment]
I can confirm the prop-type warning has now been fixed, it is sorted!

The styling is still not working as expected though.

![image](https://user-images.githubusercontent.com/4209076/73794347-56188680-47fc-11ea-8d22-9cd2a1daf834.png)

[Here is a demo which is slimmed down version of what I am building](https://codesandbox.io/s/evergreen-ui-menuitem-bug-1ynqm)

**Comment by cstrat** [comment]
Sorry to spam this one issue thread with other comments, but I found why the page is jumping around.

I am pretty sure it is to do with the popover giving the menu button focus on close. Are you be open to having a flag on the `popup` to bypass this behaviour? If so, I might try doing a PR?

https://github.com/segmentio/evergreen/blob/master/src/popover/src/Popover.js#L193

**Comment by Chrischuck** [comment]
> **edit: sorry I don't think the style fix has been applied, i saw the version bump and got excited**
> 
> I can confirm the prop-type warning has now been fixed, it is sorted!
> The styling is still not working as expected though.
> 
> ![image](https://user-images.githubusercontent.com/4209076/73794347-56188680-47fc-11ea-8d22-9cd2a1daf834.png)
> 
> [Here is a demo which is slimmed down version of what I am building](https://codesandbox.io/s/evergreen-ui-menuitem-bug-1ynqm)
> 
> _Another strange behavior I found (and is demonstrated in that codesandbox) is when closing the popup overlay the page jumps to the top. I'll look into it myself and see if I can work out the cause. To demonstrate, just open the menu dropdown, scroll to the bottom of the page, and then click to close the popup._

I split up the two prs, 1 for eslint and the other for styling. Unfortunately styling got in a little late so it'll be out in the next release, sorry about that :(

**Comment by cstrat** [comment]
All fixed in v4.23.1
